node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+    use crate::database::db_initializer::{
+        DbInitializationConfig, DbInitializer, DbInitializerReal, DATABASE_FILE,
+    };
+    use crate::database::rusqlite_wrappers::ConnectionWrapperReal;
+    use crate::database::test_utils::ConnectionWrapperMock;
+    use masq_lib::test_utils::utils::ensure_node_home_directory_exists;
+    use rusqlite::{Connection, OpenFlags};
+    use std::collections::{HashMap, HashSet};
+    use std::str::FromStr;
+
+    #[test]
+    fn insert_new_records_works() {
+        let home_dir =
+            ensure_node_home_directory_exists("failed_payable_dao", "insert_new_records_works");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .reason(PendingTooLong)
+            .checked(1)
+            .build();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let txs = vec![tx1, tx2];
+
+        let result = subject.insert_new_records(&txs);
+
+        let retrieved_txs = subject.retrieve_txs(None);
+        assert_eq!(result, Ok(()));
+        assert_eq!(retrieved_txs.len(), 2);
+        assert_eq!(retrieved_txs, txs);
+    }
+
+    #[test]
+    fn insert_new_records_throws_err_for_empty_input() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "insert_new_records_throws_err_for_empty_input",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let empty_input = vec![];
+
+        let result = subject.insert_new_records(&empty_input);
+
+        assert_eq!(result, Err(FailedPayableDaoError::EmptyInput));
+    }
+
+    #[test]
+    fn insert_new_records_throws_error_when_two_txs_with_same_hash_are_present_in_the_input() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "insert_new_records_throws_error_when_two_txs_with_same_hash_are_present_in_the_input",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let hash = make_tx_hash(123);
+        let tx1 = FailedTxBuilder::default().hash(hash).build();
+        let tx2 = FailedTxBuilder::default().hash(hash).checked(1).build();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+
+        let result = subject.insert_new_records(&vec![tx1, tx2]);
+
+        assert_eq!(
+            result,
+            Err(FailedPayableDaoError::InvalidInput(
+                "Duplicate hashes found in the input".to_string()
```

Comment:
If you wanna keep this check, you should augment the message by the whole collection of hashes supplied (of which some are duplicated).